src/include/msvc.h

```diff
@@ -34,6 +34,47 @@
 #define WT_GCC_FUNC_ATTRIBUTE(x)
 #define WT_GCC_FUNC_DECL_ATTRIBUTE(x)
 
+#define WT_UNUSED_MSVC(var) (void)(var)
+
+#define WT_ATOMIC_RELAXED 0
+#define WT_ATOMIC_ACQUIRE 1
+#define WT_ATOMIC_RELEASE 2
+#define WT_ATOMIC_ACQ_REL 3
+#define WT_ATOMIC_SEQ_CST 4
+
+#define WT_C_MEMMODEL_ATOMIC_LOAD(var, ptr, memorder) \
+    do {                                              \
+        var = *(ptr);                                 \
+        switch (memorder) {                           \
+        case WT_ATOMIC_RELAXED:                       \
+        case WT_ATOMIC_RELEASE:                       \
+            break;                                    \
+        case WT_ATOMIC_ACQUIRE:                       \
+        case WT_ATOMIC_ACQ_REL:                       \
+            WT_BARRIER();                             \
+            break;                                    \
+        case WT_ATOMIC_SEQ_CST:                       \
+            WT_FULL_BARRIER();                        \
+            break;                                    \
+        }                                             \
+    } while (0)
+
+#define WT_C_MEMMODEL_ATOMIC_STORE(ptr, value, memorder)                     \
+    do {                                                                     \
+        switch (memorder) {                                                  \
+        case WT_ATOMIC_ACQUIRE:                                              \
+        case WT_ATOMIC_ACQ_REL:                                              \
+            WT_WRITE_BARRIER();                                              \
```

Comment:
I think this should be a full barrier. Because according to the release semantic, it should prevent both read and write being reordered after the atomic store. I'm not sure for x86 if a write barrier is enough here or not.

Reply:
x86 need NO cpu barriers for acquire/release semantics, just a compiler barrier.

Reply:
Ok, understood